plugins/com.google.cloud.tools.eclipse.appengine.libraries.test/xsd/com.google.cloud.tools.eclipse.appengine.libraries.xsd

```diff
@@ -0,0 +1,267 @@
+         </xs:documentation>
+      </xs:annotation>
+      <xs:complexType>
+         <xs:choice minOccurs="1" maxOccurs="unbounded">
+            <xs:element ref="libraryFile" />
+         </xs:choice>
+         <xs:attribute name="id" type="xs:string">
+            <xs:annotation>
+               <xs:documentation>
+                  The ID of the library. The path of the classpath container defined for this library will be 
+'com.google.cloud.tools.eclipse.appengine.libraries/ID' where ID is the value defined in this attribute. 
+The classpathContainerInitializer defined to resolve the container will have the ID 
+'com.google.cloud.tools.eclipse.appengine.libraries'. The second section of the container path (the ID after the '/') 
+will determine how to resolve the entries associated with the container.
+               </xs:documentation>
+            </xs:annotation>
+         </xs:attribute>
+         <xs:attribute name="name" type="xs:string">
+            <xs:annotation>
+               <xs:documentation>
+                  A user-friendly name for the library that can be displayed in the UI, e.g.: "AppEngine Endpoints".
+               </xs:documentation>
+            </xs:annotation>
+         </xs:attribute>
+         <xs:attribute name="siteUri" type="xs:string">
+            <xs:annotation>
+               <xs:documentation>
+                  URI that can be used to display a link to the user to provide more detailed information about the library, e.g. user guides, wiki, other documentation.
+               </xs:documentation>
+            </xs:annotation>
+         </xs:attribute>
+      </xs:complexType>
+   </xs:element>
+
+   <xs:element name="libraryFile">
+      <xs:annotation>
+         <xs:documentation>
+            Describes a single artifact located in a Maven repository.
+         </xs:documentation>
+      </xs:annotation>
+      <xs:complexType>
+         <xs:sequence>
+            <xs:element ref="mavenCoordinates"/>
+            <xs:sequence minOccurs="0" maxOccurs="unbounded">
+               <xs:choice minOccurs="0" maxOccurs="1">
+                  <xs:element ref="exclusionFilter" />
+                  <xs:element ref="inclusionFilter" />
+               </xs:choice>
+            </xs:sequence>
+         </xs:sequence>
+         <xs:attribute name="export" type="xs:boolean" use="required">
+            <xs:annotation>
+               <xs:documentation>
+                  If export attribute is true, then the corresponding file will be included in the WAR file of the
+project, under WEB-INF/lib.
+               </xs:documentation>
+            </xs:annotation>
+         </xs:attribute>
+
+         <xs:attribute name="javadocUri" type="xs:anyURI">
+            <xs:annotation>
+               <xs:documentation>
+                  URI that can be used by the IDE to lookup javadoc for the classes contained in this artifact.
```

Comment:
Actually this is something I have not thought about: if no javadoc is defined, we could try to download the javadoc artifact from the Maven repository. And the same should happen with the sources.